caikit_nlp/modules/text_embedding/embedding.py

```diff
@@ -639,9 +674,39 @@ def save(self, model_path: str, *args, **kwargs):
 
 
 class SentenceTransformerWithTruncate(SentenceTransformer):
+    @staticmethod
+    def _sum_token_count(tokenized: BatchEncoding) -> int:
+        """Returns the number of non-special tokens.
+        Args:
+            tokenized: BatchEncoding
+        Returns:
+            Int total of all tokens contained in tokenized.
+            Excludes special tokens such as [CLS], [SEP], [PAD], etc.
+        """
+        error.type_check(
+            "<NLP82314993E>",
+            BatchEncoding,
+            tokenized=tokenized,
+        )
+        error.value_check(
+            "<NLP82314995E>",
+            tokenized.encodings,
+            "Number of tokenized encodings is only known when a non-python tokenizer is used",
+        )
+
+        token_count = 0
+
+        assert tokenized.encodings
```

Comment:
is this not covered by the `value_check`?

Reply:
I'll remove.

Pylance can't figure out the type narrowing with `value_check()` (that tokenized.encodings can't be `None`). Just my habit of trying to eliminate red underlines.